datamodel/create_roles.py

```diff
@@ -0,0 +1,237 @@
+        REVOKE ALL ON SCHEMA {modelname}_vl  FROM {roles[0]};
+
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_od  FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_app FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_sys FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {modelname}_vl  FROM {roles[0]};
+
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_od  REVOKE ALL ON TABLES  FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_app REVOKE ALL ON TABLES  FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_sys REVOKE ALL ON TABLES  FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {modelname}_vl  REVOKE ALL ON TABLES  FROM {roles[0]};
+        """
+    cur.execute(revoke_sql)
+    if ext_schema:
+        revoke_sql = f"""
+        REVOKE ALL ON SCHEMA {ext_schema} FROM {roles[0]};
+        REVOKE ALL ON ALL TABLES IN SCHEMA {ext_schema} FROM {roles[0]};
+        ALTER DEFAULT PRIVILEGES IN SCHEMA {ext_schema} REVOKE ALL ON TABLES  FROM {roles[0]};
+        """
+        cur.execute(revoke_sql)
+    conn.commit()
+    conn.close()
+
+
+def create_roles(
+    modelname: str,
+    pg_service: str = None,
+    grant: bool = True,
+    ext_schema: str = None,
+    db_spec_roles: Optional[bool] = False,
+):
+    """
+    Creates the roles for usage in TEKSI Modules
+    :param pg_service: the PostgreSQL service
+    :param modelname: Abbreviation of the TEKSI model
+    :param grant: Boolean defining whether to grant privileges or revoke them
+    :param ext_schema: Schema name of the extension model
+    :param db_spec_roles: will create database specific roles instead of cluster specific roles
+    """
+    if modelname and not pg_service:
+        pg_service = f"pg_{modelname}"
+
+    roles = [
+        f"{modelname}_viewer",
+        f"{modelname}_user",
+        f"{modelname}_manager",
+        f"{modelname}_sysadmin",
+    ]
+
+    if db_spec_roles:
+        db_identifier = "_" + get_db_identifier(pg_service, modelname)
+        roles = [s + db_identifier for s in roles]
+
+    in_roles = [None]
+    in_roles.extend(roles[:-1])
+
+    for role, in_role in zip(roles, in_roles):
+        create_role(pg_service, role, in_role)
+
+    if grant:
+        grant_privileges(pg_service, modelname, roles, ext_schema)
+    else:
+        revoke_privileges(pg_service, modelname, roles, ext_schema)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+
+    parser.add_argument("-m", "--modelname", help="Abbreviation of the TEKSI model")
+
+    parser.add_argument("-p", "--pg_service", help="postgres service")
+
+    parser.add_argument(
+        "-d",
+        "--database_specific_roles",
+        help="Create database specific roles",
+        default=False,
+        action=BooleanOptionalAction,
+    )
+    excl_group = parser.add_mutually_exclusive_group(required=True)
+    excl_group.add_argument(
```

Comment:
after thinking a bit more, I think this script should be renamed to `manage_roles` and use subparsers. 
This is much more convenien, futureproof and a bit more natural (`manage_roles revoke` rather than `create_roles --revoke`)

You could have 3 subparsers, `create_roles`, `grant`, and `revoke` (create_roles would call grant by default).